src/libraries/System.Text.Json/tests/Serialization/CollectionTests/CollectionTests.KeyValuePair.cs

```diff
@@ -424,15 +424,30 @@ private class ValueNameNullPolicy : JsonNamingPolicy
         [InlineData("{0")]
         [InlineData(@"{""Random"":")]
         [InlineData(@"{""Value"":1}")]
+        [InlineData(@"{null:1}")]
         [InlineData(@"{""Value"":1,2")]
         [InlineData(@"{""Value"":1,""Random"":")]
         [InlineData(@"{""Key"":1,""Key"":1}")]
+        [InlineData(@"{null:1,""Key"":1}")]
         [InlineData(@"{""Key"":1,""Key"":2}")]
         [InlineData(@"{""Value"":1,""Value"":1}")]
+        [InlineData(@"{""Value"":1,null:1}")]
         [InlineData(@"{""Value"":1,""Value"":2}")]
         public static void InvalidJsonFail(string json)
         {
             Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<KeyValuePair<int, int>>(json));
         }
+
+        [Fact]
+        public static void JsonPathIsAccurate()
+        {
+            string json = @"{""Key"":""1"",""Value"":2}";
+            JsonException ex = Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<KeyValuePair<int, int>>(json));
+            Assert.Contains("$.Key", ex.ToString());
+
+            json = @"{""Key"":1,""Value"":""2""}";
+            ex = Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<KeyValuePair<int, int>>(json));
+            Assert.Contains("$.Value", ex.ToString());
+        }
```

Comment:
If you have a wrong property "Foo", does it show up on the path?

Reply:
Following this comment, yes. Thanks.